src/lib/y2configuration_management/salt/form_builder.rb

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2configuration_management/widgets"
+
+module Y2ConfigurationManagement
+  module Salt
+    # This class builds a form according to a given specification
+    #
+    # For further information, see the forms specification at
+    # https://github.com/SUSE/spacewalk/wiki/Writing-Salt-Formulas-for-SUSE-Manager
+    class FormBuilder
+      # Build the form
+      #
+      # @param form_spec  [Y2ConfigurationManagement::Salt::Form] Form specification
+      # @param controller [Y2ConfigurationManagement::Salt::FormController] Form controller
+      # @return [Array<Y2ConfigurationManagement::Widgets::Form>]
+      def build(form_spec, controller = nil)
+        form_spec.elements.map do |element_spec|
```

Comment:
Only works with containers
```suggestion
        return build_element(element_spec, controller) unless form_spec.respond_to?(:elements)
        form_spec.elements.map do |element_spec|
```

Reply:
Yep, in my code is already an array